app/screens/channel_info/convert_private/index.js

```diff
@@ -4,24 +4,43 @@
 import {connect} from 'react-redux';
 
 import {convertChannelToPrivate} from '@mm-redux/actions/channels';
-import {General} from '@mm-redux/constants';
+import {General, Permissions} from '@mm-redux/constants';
+import {haveIChannelPermission} from '@mm-redux/selectors/entities/roles';
+import {getServerVersion} from '@mm-redux/selectors/entities/general';
 import {getCurrentChannel} from '@mm-redux/selectors/entities/channels';
+import {getCurrentTeamId} from '@mm-redux/selectors/entities/teams';
 import {getCurrentUserRoles} from '@mm-redux/selectors/entities/users';
 import {isAdmin as checkIsAdmin} from '@mm-redux/utils/user_utils';
+import {isMinimumServerVersion} from '@mm-redux/utils/helpers';
 
 import ConvertPrivate from './convert_private';
 
 function mapStateToProps(state) {
     const currentChannel = getCurrentChannel(state);
+    const channelId = currentChannel?.id || '';
+    const currentTeamId = getCurrentTeamId(state);
     const isDefaultChannel = currentChannel.name === General.DEFAULT_CHANNEL;
     const isPublicChannel = currentChannel.type === General.OPEN_CHANNEL;
     const roles = getCurrentUserRoles(state) || '';
     const isAdmin = checkIsAdmin(roles);
-    const canConvert = !isDefaultChannel && isPublicChannel && isAdmin;
+    let canConvert = !isDefaultChannel && isPublicChannel;
```

Comment:
```suggestion
    let canConvert = !isDefaultChannel && isPublicChannel && isAdmin;
```

then you don't need the else statement

Reply:
acrtually I do, cause line 28 does `canConvert = canConvert && haveIChannelPermission(` and canConvert should be true even if the user is not an admin since this permission could be added to non admin users 

I think it might make more sense to make a new variable here instead to make it more clear what the purpose is here.

something like `const isChannelConvertible = !isDefaultChannel && isPublicChannel`